client/src/components/login-signup/Login.js

```diff
@@ -1,22 +1,45 @@
-import React from "react";
+import React, { useState, useContext } from "react";
+import AuthService from '../../Services/AuthService';
+import Message from '../Message/Message';
+import {AuthContext} from '../../Context/AuthContext';
 import { useForm } from "react-hook-form";
 import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
 
-export default function Login() {
+export default function Login(props) {
   const { register, handleSubmit } = useForm();
+  const [user, setUser] = useState({ username: "", password: "" });
+  const [message, setMessage] = useState(null);
+  const authContext = useContext(AuthContext);
 
-  const onSubmit = (data) => {
-    console.log(data);
+  const onSubmit = (user, e) => {
+    e.preventDefault();
+    AuthService.login(user).then(data => {
+        const {isAuthenticated, user, message} = data;
+        if(isAuthenticated){
+          // <AuthContext.Provider value={ data } />;
+          // authContext.setIsAuthenticated(isAuthenticated);
+          //   this history object from react-router
+          // if authenticated navigate to Home  
+          props.history.push('/home')
```

Comment:
What does props.history do? Does it redirect the user to the page which is pushed in?

Reply:
for redirecting to a new page, we can use histrory object from react router using props.